hadoop-hdds/config/src/main/java/org/apache/hadoop/hdds/conf/ConfigurationSource.java

```diff
@@ -117,6 +117,17 @@ default Map<String, String> getPropsWithPrefix(String confPrefix) {
     return configMap;
   }
 
+  default Map<String, String> getPropsMatchingPrefix(String confPrefix) {
```

Comment:
nit. we could rename this to `getPropsKeepPrefix`

Reply:
I think it might be cleanest to rename the 2 similar methods in addition to adding a doc.